GameOfLife/Program.fs

```diff
@@ -0,0 +1,67 @@
+module GameOfLife.Program
+open System
+open System.Security.Cryptography
+open System.Text
+open System.Threading
+
+let rows = 15
+let columns = 15
+let timeout = 500
+let mutable runSimulation = true
+
+type Status = ``💀`` = 0 | ``😁`` = 1
+
+let private nextGeneration (currentGrid: Status [,]) = 
+    let nextGeneration = Array2D.zeroCreate rows columns
+    // Loop through every cell 
+    for row in 1..(rows-2) do
+        for column in 1..(columns-2) do
+            // find your alive neighbor
+            let mutable aliveNeighbors = 0
+            for i in -1..1 do
+                for j in -1..1 do
+                    aliveNeighbors <- aliveNeighbors + int currentGrid.[row+i, column+i]
+            let currentCell = currentGrid.[row, column]
+            // The cell needs to be subtracted 
+            // from its neighbours as it was  
+            // counted before 
+            aliveNeighbors <- aliveNeighbors - int currentCell
+            // Implementing the Rules of Life 
+            nextGeneration.[row,column] <-
+                match currentCell with
+                // Cell is lonely and dies OR Cell dies due to over population
+                | Status.``😁`` when aliveNeighbors < 2 || aliveNeighbors > 3 -> Status.``💀``
+                // A new cell is born 
+                | Status.``💀`` when aliveNeighbors = 3 -> Status.``😁``
+                // stays the same
+                | unchanged -> unchanged
+    nextGeneration
+
+let private print (future: Status[,]) =
+    let sb = StringBuilder()
+    for row in 0..(rows-1) do
+        for column in 0..(columns-1) do
+            future.[row, column] |> string |> sb.Append |> ignore
+        sb.AppendLine() |> ignore
+    Console.BackgroundColor <- ConsoleColor.Black
+    Console.CursorVisible <- false
+    Console.SetCursorPosition(0,0)
+    sb.ToString() |> Console.Write |> ignore
+    Thread.Sleep(timeout)
+
+[<EntryPoint>]
+let main _ =
+    // randomly initialize our grid
+    let mutable grid = Array2D.init rows columns (fun _ _ -> RandomNumberGenerator.GetInt32(0, 2) |> enum)
```

Comment:
F# array creation functions let you pass a lambda function to be used in initialization. Lambda's are prefixed with `fun` in F#, it's fun! Since we are randomly filling out the array, I discard the row and column arguments in the lambda. We can use the `enum` operator which is actually able to infer the enum type statically in this case!!